wasm support

Hi!

I'm trying to port a js map generation system to rust and found your awesome blue noise library.
Unfortunately, when compiling to wasm I get the error
```
  Compiling rustc-serialize v0.3.24
error[E0046]: not all trait items implemented, missing: 'encode'
    --> /Users/jh/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-serialize-0.3.24/src/serialize.rs:1358:1
     |
853  |     fn encode<S: Encoder>(&self, s: &mut S) -> Result<(), S::Error>;
     |     ---------------------------------------------------------------- 'encode' from trait
...
1358 | impl Encodable for path::Path {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing 'encode' in implementation

error[E0046]: not all trait items implemented, missing: 'decode'
    --> /Users/jh/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-serialize-0.3.24/src/serialize.rs:1382:1
     |
904  |     fn decode<D: Decoder>(d: &mut D) -> Result<Self, D::Error>;
     |     ----------------------------------------------------------- 'decode' from trait
...
1382 | impl Decodable for path::PathBuf {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing 'decode' in implementation
```
It seems to be a problem with depending on serialization, would it be possible to put whatever features use serialization behind a feature gate?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ alga = "0.7"
 num-traits = "0.2"
 lazy_static = "0.1"
 modulo = "0.1"
-sphere = "0.2"
+sphere = "0.3"
 
 [dev-dependencies]
 nalgebra = "0.16"

